File: app/src/main/java/com/vladiyak/cryptocurrencyapp/utils/CustomMarkerView.java
package com.vladiyak.cryptocurrencyapp.utils;

import android.content.Context;
import android.widget.TextView;

import com.github.mikephil.charting.components.MarkerView;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.highlight.Highlight;
import com.github.mikephil.charting.utils.MPPointF;
import com.vladiyak.cryptocurrencyapp.R;

import java.text.SimpleDateFormat;
import java.util.Date;

public class CustomMarkerView extends MarkerView {

    private TextView textViewEntry;

    public CustomMarkerView(Context context, int layoutResource) {
        super(context, layoutResource);
        // this markerview only displays a textview
        textViewEntry = (TextView) findViewById(R.id.textViewEntry);
    }

//     callbacks everytime the MarkerView is redrawn, can be used to update the
//     content (user-interface)
    @Override
    public void refreshContent(Entry e, Highlight highlight) {
        // format date
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd, HH:mm");
        String dateValue = String.format("%.0f", e.getX());
        Date date = new Date(Long.valueOf(dateValue));
        String formattedDate = sdf.format(date);
        // set the entry-value as the display text
        textViewEntry.setText("$" + e.getY() + "\n" + formattedDate);
        // this will perform necessary layouting
        super.refreshContent(e, highlight);
    }

    private MPPointF mOffset;

    @Override
    public MPPointF getOffset() {
        if (mOffset == null) {
            // center the marker horizontally and vertically
            mOffset = new MPPointF(-(getWidth() / 2), -2 * getHeight());
        }
        return mOffset;
    }
}

File: app/src/main/java/com/vladiyak/cryptocurrencyapp/utils/CustomMarkerView.kt
package com.vladiyak.cryptocurrencyapp.utils

import android.annotation.SuppressLint
import android.content.Context
import android.widget.TextView
import com.github.mikephil.charting.components.MarkerView
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.highlight.Highlight
import com.github.mikephil.charting.utils.MPPointF
import com.vladiyak.cryptocurrencyapp.R
import java.text.SimpleDateFormat
import java.util.Date

@SuppressLint("ViewConstructor")
class CustomMarkerView (context: Context, layoutResource: Int) : MarkerView(context, layoutResource) {

    private var mOffset: MPPointF? = null
    val textViewDate = findViewById<TextView>(R.id.tvMarkerDate)
    val textViewValue = findViewById<TextView>(R.id.tvMarkerValue)

    override fun refreshContent(e: Entry?, highlight: Highlight?) {
        val sdf = SimpleDateFormat("yyyy-MM-dd\nHH:mm:ss")
        val dateValue = String.format("%.0f", e!!.x)
        val date = Date(java.lang.Long.valueOf(dateValue))
        val formattedDate = sdf.format(date)

        textViewDate.text = formattedDate
        textViewValue.text = "$" + e!!.y

        super.refreshContent(e, highlight)
    }

    override fun getOffset(): MPPointF {
        if (mOffset == null) {
            mOffset = MPPointF((-(width / 2.1)).toFloat(), (-height).toFloat())
        }
        return mOffset as MPPointF
    }

}